extensions.html

```diff
@@ -649,6 +649,168 @@ <h2>
         </p>
       </dd>
     </section>
+    <section>
```

Comment:
Let's define extended buttons in the main spec document (index.html). The extensions document is not officially part of the spec and we want to eventually remove it.

The spec PR should extend [GamepadButton](https://www.w3.org/TR/gamepad/#gamepadbutton-interface) to add a new attribute (perhaps called `GamepadButton.name`) and an algorithm for the getter. The type of the new attribute will be a new enum (perhaps called `GamepadButtonName`). The enum type should be defined in the spec along with all its values.

Add steps to initialize the new attribute when [initializing buttons](https://www.w3.org/TR/gamepad/#dfn-initializing-buttons). The steps should define what to do if a gamepad is missing a button defined in the Standard Gamepad. I think it makes sense to use a special value for these buttons so applications can identify which buttons are not actually mapped. (Ideally we'd set the button to null, but this would break existing apps which don't check for null.)

Also, initialization steps should define what to do if a gamepad has a button that doesn't have a predefined Standard Gamepad index or enum value. Do we enumerate it? If so, what should the new attribute's value be? Perhaps we can add a special "unrecognized" value for this.